### ISSUE TITLE
[BUG] Serviceaccount not found error

### ISSUE THREAD

**Issue by sai1621** [stackrox/kube-linter#137]
**System info:**
 - MacOS

**Describe the bug**
Throwing a `non-existent-service-account` error even when the serviceaccount exists

**To Reproduce**

1. Create a serviceaccount : 

`kubectl create sa lint-test-sa`

2. Create a deployment.yaml file with the content below :

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: kube-linter-test
spec:
  replicas: 1
  selector:
    matchLabels:
      label1: content
  template:
    metadata:
      labels:
        label1: content
    spec:
      serviceAccountName: lint-test-sa
      containers:
        - name: kube-linter-test-container
          image: "image/image:tag"
```

3. Run the `kube-linter` :

`kube-linter templates deployment.yaml`

```
templates.yaml: (object: <no namespace>/kube-linter-test apps/v1, Kind=Deployment) serviceAccount "lint-test-sa" not found (check: non-existent-service-account, remediation: Make sure to create the service account, or to refer to an existing service account.)
```

**Expected behavior**
The linter should show no errors. I am using kube-linter version 0.1.6.

**Comment by viswajithiii** [comment]
Hi @sai1621, thanks for filing this issue! This is expected behaviour. The linter is looking for the YAML file corresponding to that service account, and doesn't actually connect to the cluster. If your service account manifest is not stored with the rest of your manifests, it will throw an error; if you generally create service accounts using `kubectl create sa` then it would make sense for you to disable this check. In general, though, I would recommend using a YAML file for the service account and storing that alongside the deployment YAML file. This way, you can ensure that a rename in one is reflected in the other using the `kube-linter` check.

Does that make sense?

**Comment by johnrwat** [comment]
Hi @viswajithiii - I've managed to test your solution above, using a service account defined within the same file which worked no problem. E.g.:

`$ kube-linter version
0.1.6-8-ga8478eef70`

Using call:  kube-linter lint <file-contents-below>.yaml

`---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: example-service-account
  namespace: dbs

---
apiVersion: v1
kind: Pod
metadata:
  name: static-web
  namespace: dbs
  labels:
    role: myrole
    owner: firstname.lastname@example.com
  annotations:
    email: john.watson@example.com
spec:
  serviceAccountName: example-service-account
  containers:
    - name: web
      image: nginx
      livenessProbe:
        exec:
          command:
            - cat
            - /tmp/healthy
        initialDelaySeconds: 5
        periodSeconds: 5
      readinessProbe:
        exec:
          command:
            - cat
            - /tmp/healthy
        initialDelaySeconds: 5
        periodSeconds: 5
      securityContext:
        runAsNonRoot: true
        readOnlyRootFilesystem: true
      resources:
        requests:
          memory: "2048Mi"
          cpu: "2"
        limits:
          memory: "4096Mi"
          cpu: "2"
      ports:
        - name: web
          containerPort: 80
          protocol: TCP
`
**The linter throws:**
`No lint errors found!`

I had some issues getting this to work because the namespace within my ServiceAccount object was originally not defined.
I.e. using the file contents of:
`apiVersion: v1
kind: ServiceAccount
metadata:
  name: example-service-account`
 
Rather than:
`apiVersion: v1
kind: ServiceAccount
metadata:
  name: example-service-account
  namespace: dbs`

**The linter throws:**
`deployment.yml: (object: dbs/static-web /v1, Kind=Pod) serviceAccount "example-service-account" not found (check: non-existent-service-account, remediation: Make sure to create the service account, or to refer to an existing service account.)

Error: found 1 lint error`

It might be worth adding an improvement for the missing namespace attribute in general, or by doing some kind of cross-checking for a ServiceAccount in `any` namespace. 

I'd say based on the above though, you could close this bug.

**Comment by viswajithiii** [comment]
Hi @johnrwat, yes, that's a valid observation, and something we need to fix. It's being tracked in #140. Thanks for the detailed example!

**Comment by akdes** [comment]
We have found another problem for this issue: 

As soon as you create your helm-chart folder with subfolders like 
myHelmChart/frontend/...yaml -> points to the central service account
myHelmChart/backend/...yaml -> also points to the central service account
myHelmChart/serviceaccount.yaml

You receive the same error as described.
